ios/FluentUI.Demo/FluentUI.Demo/Demos/DateTimePickerDemoController.swift

```diff
@@ -87,36 +126,65 @@ class DateTimePickerDemoController: DemoController {
+        let (startDate, endDate, _) = calcDatePickerParams()
         dateTimePicker.present(from: self, with: .dateRange, startDate: startDate, endDate: endDate, datePickerType: datePickerType)
     }
 
     @objc func presentTabbedDateRangePicker() {
-        let startDate = self.startDate ?? Date()
-        let endDate = self.endDate ?? Calendar.current.date(byAdding: .day, value: 1, to: startDate) ?? startDate
+        let (startDate, endDate, _) = calcDatePickerParams()
         dateTimePicker.present(from: self, with: .dateRange, startDate: startDate, endDate: endDate, datePickerType: datePickerType, dateRangePresentation: .tabbed)
     }
 
     @objc func presentDateTimeRangePicker() {
-        let startDate = self.startDate ?? Date()
-        let endDate = self.endDate ?? Calendar.current.date(byAdding: .hour, value: 1, to: startDate) ?? startDate
+        let (startDate, endDate, _) = calcDatePickerParams()
         dateTimePicker.present(from: self, with: .dateTimeRange, startDate: startDate, endDate: endDate, datePickerType: datePickerType)
     }
 
     @objc func presentCustomSubtitlePicker() {
-        let startDate = self.startDate ?? Date()
-        let endDate = self.endDate ?? Calendar.current.date(byAdding: .day, value: 1, to: startDate) ?? startDate
-        let titles: DateTimePicker.Titles
-        if datePickerType == .calendar && !UIAccessibility.isVoiceOverRunning {
-            titles = .with(startSubtitle: "Assignment Date", endSubtitle: "Due Date")
-        } else {
-            titles = .with(startTab: "Assignment Date", endTab: "Due Date")
-        }
+        let (startDate, endDate, titles) = calcDatePickerParams()
         dateTimePicker.present(from: self, with: .dateRange, startDate: startDate, endDate: endDate, datePickerType: datePickerType, titles: titles)
     }
 
+    @objc func presentPickerLeftToolbarView() {
+        let (startDate, endDate, titles) = calcDatePickerParams()
+        let leftBarButtonItem = DemoBarButtonItems.cancel(target: self, action: #selector(handleDidTapCancel))
+        dateTimePicker.present(from: self, with: .dateRange, startDate: startDate, endDate: endDate, datePickerType: datePickerType, titles: titles, leftBarButtonItem: leftBarButtonItem)
+    }
+
+    @objc func presentPickerLeftRightToolbarView() {
+        let (startDate, endDate, titles) = calcDatePickerParams()
+        let leftBarButtonItem = DemoBarButtonItems.confirm(target: self, action: #selector(handleDidTapDone))
+        let rightBarButtonItem = DemoBarButtonItems.cancel(target: self, action: #selector(handleDidTapCancel)) // or simply assign UIBarButtonItem() to hide the default confirm button
+        dateTimePicker.present(from: self, with: .dateRange, startDate: startDate, endDate: endDate, datePickerType: datePickerType, titles: titles, leftBarButtonItem: leftBarButtonItem, rightBarButtonItem: rightBarButtonItem)
+    }
+
+    @objc private func handleDidTapCancel() {
+        self.dateTimePicker.dismiss()
```

Comment:
nit: you shouldnt need "self" here and other places

Reply:
Removed all unnecessary 'self'

Reply:
Removed all unnecessary 'self'